add_setup_py: Support for creating setup.py

jokva opened the pull request:
setuptools/setup.py is a useful, but peculiar, tool for building python
packages that is well integration with PyPI. It's a build system of its
own, and for some projects that essentially means double bookeeping of
build scripts.

CMake has a lot of information on what is to be built and how, so it's
fairly natural to want to use this build information to create a
setup.py that can build and upload this package as a step in an
automated build+deploy process.

add_setup_py does exactly this, assuming add_python_package is used and
the base cmakelists is reasonably well set up.

jokva committed:
add_setup_py: Support for creating setup.py

setuptools/setup.py is a useful, but peculiar, tool for building python
packages that is well integration with PyPI. It's a build system of its
own, and for some projects that essentially means double bookeeping of
build scripts.

CMake has a lot of information on what is to be built and how, so it's
fairly natural to want to use this build information to create a
setup.py that can build and upload this package as a step in an
automated build+deploy process.

add_setup_py does exactly this, assuming add_python_package is used and
the base cmakelists is reasonably well set up.

kjellkongsvik commented on PythonPackage.cmake:
```diff
@@ -79,10 +102,109 @@ else()
+#
+# All properties are lists, and the content correspond to the non-namespaced
+# properties (includes, sources etc.)
+function(include_target_dependencies pkg tgt depend_dirs)
```
prefiix namespace

kjellkongsvik commented on PythonPackage.cmake:
```diff
@@ -13,20 +13,12 @@
 #  for more details
 
 # This module exports the following functions:
-# * to_path_list(var path)
```
No doc for pycmake_to_path_list?

jokva replied:
No, because it's not public (anymore).

kjellkongsvik commented on PythonPackage.cmake:
```diff
@@ -79,10 +102,109 @@ else()
+endfunction ()
+
+# internal. Traverse the tree of dependencies (linked targets) that are actual
+# cmake targets an add to a list
```
typo: an -> and

jokva replied:
Fix'd

jokva committed:
Install from build dir, not source dir

Since pycmake potentially relocates, reorganises and adds version string
to the python package, the install should happen from the build
directory, not the source directory. This makes configurations and
structure carry over to the installed package.
